podcast/2.16.md

```diff
@@ -0,0 +1,1131 @@
+
+But Twitter, of course, had theirs, Pants. Has theirs, still, I believe. I don't know, I haven't checked in to see if they've moved to anything else since I left. We were on Maven when I started. Maven was causing us a lot of trouble, a lot of trouble. It couldn't really walk the dependency graph in any reasonable way, and you really need that once you're in a mono repo. That's almost job one for your build tool on a mono repo. We had these crazy ass Maven POM XMLs that were deeply nested.
+
+I remember buying a Mac Pro, like the trash can black Mac Pro, because I needed something that could literally compile the whole repo for reasons and I needed enough memory. And so the right choice was, of course, to spend, I don't know how much, on a really expensive computer from Apple. I don't even think I expensed it.
+
+**Beyang Liu:**
+
+Wow.
+
+**James Waldrop:**
+
+Yeah. Well, I still have it so I get my money's worth. At any rate, yeah, it was a hard problem. And so, Pants was our solution to that problem.
+
+**Beyang Liu:**
+
+Interesting. I guess taking a step back, I bet a good number of listeners are working on developer productivity, or are maybe part of a developer productivity team, or maybe even running it. What are your thoughts on how to run an effective dev prod organization? What are the things to look out for? What are the things that you measure? How do you define success?
+
+**James Waldrop:**
+
+It's good that you mentioned measurement. Measurement is a two-edged sword.
+
+If you're on a team that has the goal to name itself “developer productivity,” you've got to be able to show that you're making people more productive, right?
+
+**Beyang Liu:**
+
+Yeah.
+
+**James Waldrop:**
+
+If you're not doing that, then you're not going to get headcount, that's for sure. Headcount is lifeblood for these teams. So, you need some metrics, but you really need to avoid being overly managed by your metrics. It can be very easy for those metrics to become targets that you have to hit as opposed to simply: are we improving life for developers?
+
+So, we measured things like build times, obviously. We knew all the slowest build targets. We could identify whether or not your machine needed to be upgraded because build targets were running slow for you, that we're running fast for other people. We could say, "Huh, there's something going on with your system. Let's figure out what it is." It's a corporate environment, so we could log all of the builds and ship them all up to our servers and see our metrics change as we release things. People loved that.
```

Comment:
```suggestion
So, we measured things like build times, obviously. We knew all the slowest build targets. We could identify whether or not your machine needed to be upgraded because build targets were running slow for you, that were running fast for other people. We could say, "Huh, there's something going on with your system. Let's figure out what it is." It's a corporate environment, so we could log all of the builds and ship them all up to our servers and see our metrics change as we release things. People loved that.
```